hawtio-war/src/main/webapp/WEB-INF/web.xml

```diff
@@ -245,11 +245,6 @@
     <listener-class>io.hawt.HawtioContextListener</listener-class>
   </listener>
 
-  <error-page>
-    <error-code>404</error-code>
-    <location>/index.html</location>
-  </error-page>
-
```

Comment:
This was added by Jimmi in 2014 in 1a5c3e48600abaa87f2bd3a41ac1e490d4890168. Initially I thought it's important, but then I found we're getting proper `index.html` page but with 404 error, which is simply confusing and _may_ affect browser in some future version (like not rendering).

And I really spent last ~2 weeks refactoring this and thinking about it at different angles.
So the reason of this commit was: if user hits CTRL-R, F5 or any kind of browser refresh, current URL (which is ALWAYS a client-side router URL or (somehow) index.html) - and this is what I've carefully handled in LoginRedirectServlet renamed to ClientRouteRedirectServlet.

So you can CTRL-R anything and:
* if it's static resource (image/js/css/index.html/favicon.ico/hawtconfig.json/...), it's served
* it can't be `/jolokia` because it's handled by POST, but if it is (like http://localhost:8080/hawtio/jolokia/version), you'll get JSON
* if it's `/login` (a _special_ client-side route URL), it's handled specially depending on whether authentication is enabled or not and whether you're authenticated or not.

Reply:
OK, then let's remove this item now.

Note that now we then need to provide our own 404.html because the default 404 page from a web container may expose some sensitive information and the security best practice always advises us to provide our own.